dbt/adapters/snowflake/relation.py

```diff
@@ -33,3 +51,48 @@ def is_dynamic_table(self) -> bool:
     @classproperty
     def DynamicTable(cls) -> str:
         return str(SnowflakeRelationType.DynamicTable)
+
+    @classproperty
+    def get_relation_type(cls) -> Type[SnowflakeRelationType]:
+        return SnowflakeRelationType
+
+    @classmethod
+    def from_runtime_config(cls, runtime_config: RuntimeConfigObject) -> RelationConfigBase:
+        model_node: ModelNode = runtime_config.model
+        relation_type = SnowflakeRelationType(model_node.config.materialized)
+
+        if relation_config := cls.relation_configs.get(relation_type):
+            return relation_config.from_model_node(model_node)
+
+        raise DbtRuntimeError(
+            f"from_runtime_config() is not supported for the provided relation type: {relation_type}"
+        )
+
+    @classmethod
+    def dynamic_table_config_changeset(
+        cls,
+        dynamic_table: SnowflakeDynamicTableConfig,
+        relation_results: RelationResults,
+    ) -> Optional[SnowflakeDynamicTableConfigChangeset]:
+        config_changeset = SnowflakeDynamicTableConfigChangeset()
+
+        existing_dynamic_table = SnowflakeDynamicTableConfig.from_relation_results(
+            relation_results
+        )
+        assert isinstance(existing_dynamic_table, SnowflakeDynamicTableConfig)
```

Comment:
Leftover `assert`? Should it be an exception?

Reply:
This is to make the linter happy. `from_relation_results()` returns a generic `RelationConfigBase()` because it's on the base class. The base class has no platform-specific attributes like `target_lag`, `warehouse`, etc. Instead of doing a `# type: ignore` and dealing with the warnings in the IDE, I solve both with this assert.

Reply:
So what happens when this crashes on a user for some reason down the line against all our assumptions? We fine with this being the possible bug report?

Reply:
Yeah, I should probably wrap that in a `try` / `catch` and make it pretty, fair point.